web-demo-app/src/app/board/submodule.ts

```diff
@@ -0,0 +1,30 @@
+import { BoardModule } from './board.module';
+import { Credentials } from '../credentials';
+import { Session } from '@closerplatform/closer-sdk';
+import { SpinnerClient } from '@swagger/spinner';
+
+export abstract class SubModule {
+  public abstract readonly NAME: string;
+  protected inner: JQuery;
+
+  constructor (public boardModule: BoardModule, public credentials: Credentials) { }
+
+  public abstract init(session: Session, spinnerClient: SpinnerClient): Promise<void>;
+
+  public toggleVisible = async (visible = true): Promise<void> => {
+    if (visible) {
+      if (this.inner) {
+        await this.onShow();
+        this.inner.show();
+      }
+    } else {
+      if (this.inner) {
+        await this.onHide();
+        this.inner.hide();
+      }
+    }
+  }
+
+  protected onShow = (): Promise<void> => Promise.resolve();
+  protected onHide = (): Promise<void> => Promise.resolve();
```

Comment:
To chyba nie musi być Promise? Domyślna implementacja też nie wygląda na konieczną

Reply:
Dałem promisa, bo w conversation.module onshow używa await, 
domyślna implementacja jest, ponieważ tą funkcję możemy nadpisać w klasach dziedziczących tylko jeśli trzeba coś zrobić pokazując/ukrywając element,
mogę zmienić, że onShow/ onHide może być funkcją albo domyślnie undefined, np.
  protected onShow: (() => Promise<void>) | undefined = undefined;